### PR TITLE
clarify the routing component documentation a bit

### PR DESCRIPTION
**Pull request by dbu**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | Relates to #5709 |

### CODE REVIEW

File: components/routing/introduction.rst [hunk 1]
```diff
@@ -62,11 +62,15 @@ URL path and some array of custom variables in its constructor. This array
 of custom variables can be *anything* that's significant to your application,
 and is returned when that route is matched.
 
-If no matching route can be found a
-:class:`Symfony\\Component\\Routing\\Exception\\ResourceNotFoundException` will be thrown.
+The :method:``UrlMatcher::match() <Symfony\\Component\\Routing\\UrlMatcher::match``
```
Comment (xabbuh) [inline components/routing/introduction.rst]: missing angle bracket

Comment (WouterJ) [inline components/routing/introduction.rst]: and double backticks should be single ones

**Comment by dbu** [comment]
fixed

**Comment by xabbuh** [comment]
:+1:

**Comment by dbu** [comment]
anything left to do here or can this be merged?

**Comment by xabbuh** [comment]
@javiereguiluz @weaverryan @WouterJ What do you think?

File: components/routing/introduction.rst [hunk 2]
```diff
@@ -62,11 +62,15 @@ URL path and some array of custom variables in its constructor. This array
...
-:class:`Symfony\\Component\\Routing\\Exception\\ResourceNotFoundException` will be thrown.
+The :method:`UrlMatcher::match() <Symfony\\Component\\Routing\\UrlMatcher::match>`
+returns the variables you set on the route as well as the wildcard placeholders
+(see below). Your application can now use this information to continue
+processing the request. In addition to the configured variables, a ``_route``
+key is added, which holds the name of the matched route.
 
-In addition to your array of custom variables, a ``_route`` key is added,
-which holds the name of the matched route.
+If no matching route can be found during ``match``, a
```
Comment (javiereguiluz) [inline components/routing/introduction.rst]: I'd remove `during match` because it looks redundant.

Comment (dbu) [inline components/routing/introduction.rst]: fixed

**Comment by weaverryan** [comment]
Nice job @dbu, thanks :)

**Pull request merged**